src/bootstrap/doc.rs

```diff
@@ -537,7 +537,18 @@ impl Step for Rustc {
 
     fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
         let builder = run.builder;
-        run.krate("rustc-main").default_condition(builder.config.docs)
+
+        let mut paths = Vec::new();
+        for entry in fs::read_dir(builder.src.join("compiler")).expect("read_dir failed") {
```

Comment:
`krate("rustc-main")` should already do pretty much just this I believe, but without needing to read the directory (it relies on cargo metadata).

Reply:
It doesn't allow to do `x.py compiler` or `x.py compiler/rustc_pretty_ast` unfortunately... Which is why I added this way too.

Reply:
The first of those I get, but seems like that's solved with a .path("compiler"), not this -- can you clarify the latter though? I would expect that to work (and it seems to work elsewhere -- e.g., the Std doc step has `all_krates("test")`)

Reply:
You are absolutely right. I simplified it as you suggested!